lib/spack/spack/cray_manifest.py

```diff
@@ -58,6 +80,16 @@ def compiler_from_entry(entry):
     compiler_cls = spack.compilers.class_for_compiler_name(compiler_name)
     spec = spack.spec.CompilerSpec(compiler_cls.name, version)
     paths = [paths.get(x, None) for x in ("cc", "cxx", "f77", "fc")]
+
+    if missing_paths:
+        tty.warn(
```

Comment:
This should behave the same:
```suggestion
        warnings.warn(
```
but will use stdlib modules only. The pro in doing that is that we'll control how to show warnings from a single place:

https://github.com/spack/spack/blob/a236fce31f5d5018c788f51cda50492f1f241cba/lib/spack/spack/main.py#L562-L570